src/scripts/mentoring-data-migration.js

```diff
@@ -0,0 +1,1325 @@
+			const availableUpdateColumns = updateColumns.filter((col) => existingColumns.includes(col))
+
+			if (availableUpdateColumns.length === 0) {
+				console.log(`⚠️  Table ${name} has no target columns, skipping`)
+				return
+			}
+
+			// Only update tenant_code for user_id tables
+			if (availableUpdateColumns.includes('tenant_code')) {
+				const citusEnabled = await this.isCitusEnabled()
+
+				// Undistribute if needed
+				let wasDistributed = false
+				if (citusEnabled && hasPartitionKey) {
+					wasDistributed = await this.undistributeTable(name)
+				}
+
+				try {
+					await this.executeUserIdUpdates(name)
+
+					// Add detailed logging like in migration
+					await this.logTableUpdateResults(name)
+
+					// Redistribute if needed
+					if (citusEnabled && wasDistributed) {
+						await this.redistributeTable(name)
+					}
+				} catch (error) {
+					console.error(`❌ Error during updates for ${name}:`, error.message)
+					throw error
+				}
+			}
+
+			console.log(`✅ Completed ${name}`)
+		} catch (error) {
+			console.error(`❌ Error processing table ${name}:`, error)
+			throw error
+		}
+	}
+
+	/**
+	 * Execute updates for tables with user_id using user_extensions data
+	 */
+	async executeUserIdUpdates(tableName) {
+		console.log(`📊 Executing user_id updates for ${tableName}...`)
+
+		const transaction = await this.sequelize.transaction()
+
+		try {
+			// Get table configuration
+			const tableConfig = this.tablesWithUserId.find((t) => t.name === tableName)
+
+			let updateQuery
+
+			if (tableConfig.useSessionLookup) {
+				// Special case for post_session_details: get data from sessions table based on session_id
+				const sessionIdColumn = tableConfig.columns.session_id
+
+				// Build SET clause based on available update columns
+				const setClauses = []
+				if (tableConfig.updateColumns.includes('tenant_code')) {
+					setClauses.push('tenant_code = s.tenant_code')
+				}
+				if (tableConfig.updateColumns.includes('organization_code')) {
+					setClauses.push('organization_code = s.organization_code')
+				}
+				setClauses.push('updated_at = NOW()')
+
+				// Build WHERE conditions based on what columns sessions table actually has
+				// Check for non-default values since migration sets defaults
+				const defaultTenantCode = process.env.DEFAULT_ORGANISATION_CODE || 'default'
+				const whereConditions = [
+					`${tableName}.${sessionIdColumn} = s.id`,
+					`s.tenant_code IS NOT NULL AND s.tenant_code != '${defaultTenantCode}'`,
+				]
+
+				// Only add organization_code condition if we're actually using it
+				if (tableConfig.updateColumns.includes('organization_code')) {
+					whereConditions.push('s.organization_code IS NOT NULL')
+				}
+
+				// Use batch processing for session lookup - get distinct session organizations
+				const [sessionOrgs] = await this.sequelize.query(`
+				SELECT DISTINCT s.organization_id::text as org_id
+				FROM sessions s
```

Comment:
@sumanvpacewisdom join the tables b/w postSession, session and user ext  the loop on the data update the record